expirationd.lua

```diff
@@ -327,48 +390,202 @@ end
+--     Function to call before starting a full scan iteration. Default value
+--     is a function that do nothing.
+-- @func[opt] options.on_full_scan_success
+--     Function to call after successfully completing a full scan iteration.
+--     Default value is a function that do nothing.
+-- @func[opt] options.process_expired_tuple
+--     Applied to expired tuples, receives `space_id`, `args`, `tuple` as
+--     arguments. When `process_expired_tuple` is not passed (or `nil` passed),
+--     tuples are removed.
+--
+-- Example of function:
+--
+-- ```
+-- local function put_tuple_to_archive(space_id, args, tuple)
+--     box.space[space_id]:delete{tuple[1]}
+--     local email = get_field(tuple, 2)
+--     if args.archive_space_id ~= nil and email ~= nil then
+--         box.space[args.archive_space_id]:replace{email, fiber.time()}
+--     end
+-- end
+-- ```
+--
+-- @func[opt] options.process_while
+--     Function to call before checking each tuple. If it returns false, the
+--     task will stop until next full scan. Default is a function that always
+--     return `true`.
+--
+-- Example of function:
+--
+-- ```
+-- local function process_while()
+--     return false
+-- end
+-- ```
+--
+-- @string[opt] options.start_key
+--     Start iterating from the tuple with this index value. Or when iterator
+--     is 'EQ', iterate over tuples with this index value. Must be a value of
+--     the same data type as the index field or fields, or a function which
+--     returns such value. If omitted or nil, all tuples will be checked.
+-- @number[opt] options.tuples_per_iteration
+--     Number of tuples to check in one batch (iteration). It is allowed to
+--     pass an FFI number: `1LL`, `1UL` etc. Default value is 1024.
+-- @number[opt] options.vinyl_assumed_space_len_factor
+--     Factor for recalculation of vinyl space size. Vinyl space size can't be
+--     counted (since many operations, `upsert` for example, are applied when
+--     you address some data), so you should count (approximate space size)
+--     tuples with the first start. `vinyl_assumed_space_len` is approximate
+--     count for first run and `vinyl_assumed_space_len_factor` for next
+--     milestone (after we've reached next milestone is `*` and so on). It is
+--     allowed to pass an FFI number: `1LL`, `1UL` etc. Default value is 2.
+-- @number[opt] options.vinyl_assumed_space_len
+--     Assumed size of vinyl space (in the first iteration).
+--     Vinyl space size can't be counted (since many operations, `upsert` for
+--     example, are applied when you address some data), so you should count
+--     (approximate space size) tuples with the first start.
+--     `vinyl_assumed_space_len` is approximate count for first run and
+--     `vinyl_assumed_space_len_factor` for next milestone (after we've reached
+--     next milestone is `*` and so on). It is allowed to pass an FFI number:
+--     `1LL`, `1UL` etc. Default value is 10^7.
+--
+-- @treturn task_obj
```

Comment:
The link goes to the task function. Is it correct? I would expect that it'll go to the list of task object methods.

Reply:
I keep it as is because I don't get how to fix it in LDoc.